draft-barnes-acme.md

```diff
@@ -1093,6 +1093,26 @@ type (required, string):
 To validate a DNS challenge, the server queries for TXT records under the validation domain name.  If it receives a record whose contents match the token in the challenge, then the validation succeeds.  Otherwise, the validation fails.
 
 
+## Wildcard Domain Validation
+
+When the identifier being validated is a domain name, it may contain a wildcard character "\*" as its left-most label (e.g., "\*.foo.example.com"). A wildcard certificate contains a wildcard domain in a CN or subjectAltName of type DNS-ID and automatically vouches for any and all host names within the bare domain name (the FQDN to the immediate right of the wildcard label).
+
+Domain validation for issuance of a wildcard certificate proceeds as follows:
+1. The server MUST validate that the identifier contains a single wildcard character and that the wildcard character is the left-most label. For instance, "foo.\*.example.com" is invalid.
+2. The server MUST validate that the wildcard label does not fall immediately to the left of a registry-controlled or public suffix. For instance, "\*.co.uk" is invalid.
+3. The server SHOULD validate that there exists a valid DNS A, AAAA, or CNAME record for the identifier. For instance, if the identifier being validated is "\*.example.com", the following record would suffice:
```

Comment:
I think that this would be untenable for a great many deployments. I believe that a better test would be to prove that control over a name or names of the server's choice can be demonstrated. That might use the usual methods.

You might like to allow a client to exclude a small number of in-use domains from this, but I haven't thought that through.